common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Strategy.java

```diff
@@ -49,23 +49,26 @@ protected void validateTokenRequest(TokenRequest request) {
      */
     @Override
     public Account createAccount(TokenResponse response) {
-        IDToken idToken = new IDToken(response.getIdToken());
-        ClientInfo clientInfo = new ClientInfo(((AzureActiveDirectoryTokenResponse) response).getClientInfo());
+        IDToken idToken = null;
+        ClientInfo clientInfo = null;
+        try {
+            idToken = new IDToken(response.getIdToken());
```

Comment:
I'm wondering whether it makes more sense to  validate the entire response prior to attempting to use it in various classes.   In that way... we avoid failing at some point during the processing of the response.  And we avoid the need to handle exceptions all over the place related to parsing, etc....  We validate/parse it in a single step such that it's prepared for further use.